src/scanner-2/scan-util.ts

```diff
@@ -587,6 +647,11 @@ export function scanOthers (state : LexerState) : TokenKind {
 export function peekTokenAfterExtras (state : LexerState) : TokenKind {
     const tmp = state.clone();
     while (!tmp.isEof(0)) {
+        /**
+         * @todo Use a `tryScanExtra()` function.
+         * Using `scan()` will result in recursive `scan()` calls
+         * whenever we encounter a keyword.
```

Comment:
If we have n consecutive keywords, we end up calling `peekTokenAfterExtras()`

1 + 2 + 3 + ... + n = O(n^2) times

This is bad.